engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -428,22 +432,38 @@ public Optional<Module> findModuleByExpressionId(UUID expressionId) {
   /**
    * Creates a class path. A class path is capable to load Java classes in an isolated manner.
    *
-   * @param files files to register
+   * @param pkg package to find class path for
    */
   @TruffleBoundary
-  public synchronized EnsoClassPath findClassPath(Package<?> pkg) {
+  public synchronized EnsoClassPath findClassPath(Package<TruffleFile> pkg) {
     var l = classPaths.get(pkg);
     if (l == null) {
-      l = new EnsoClassPath();
+      var ch = pkg.polyglotDir().resolve("java");
+      l = new EnsoClassPath(ch);
       classPaths.put(pkg, l);
     }
     return l;
   }
 
   public final class EnsoClassPath {
-    private final HostClassLoader hostClassLoader = new HostClassLoader();
-
-    private EnsoClassPath() {}
+    private final ModuleLayer.Controller cntrl;
+    private final ModuleLayer layer;
+    private final HostClassLoader hostClassLoader;
+
+    private EnsoClassPath(TruffleFile file) {
+      var parent = ModuleLayer.boot();
+      var parentCfgs = Collections.singletonList(parent.configuration());
+      var parentModules = Collections.singletonList(parent);
+      var parentLoader = parent.findLoader("java.base");
+      var finder = ModuleFinder.of(Paths.get(file.toUri()));
+      var cfg = Configuration.resolve(finder, parentCfgs, finder, Collections.emptyList());
+
+      this.hostClassLoader = new HostClassLoader();
+
+      this.cntrl = ModuleLayer.defineModulesWithOneLoader(cfg, parentModules, parentLoader);
+      this.layer = cntrl.layer();
+      System.out.println("modules: " + layer.modules());
```

Comment:
Whatever I do, I am getting empty `[]` here. Probably I should specify something else than `emptyList()` in the `cfg` definition. But I don't know any module name to ask for! Ideas, @Akirathan?

Reply:
Using `ModuleFinder.findAll` per Pavel's suggestion in 8700cc8141